crates/topos-p2p/src/runtime/handle_event/discovery.rs

```diff
@@ -35,74 +32,106 @@ impl EventHandler<Box<Event>> for Runtime {
             }
 
             Event::OutboundQueryProgressed {
-                result: QueryResult::Bootstrap(res),
                 id,
-                ..
-            } => {
-                debug!("BootstrapResult query: {id:?},  {res:?}");
+                result:
+                    QueryResult::Bootstrap(BootstrapResult::Ok(BootstrapOk {
+                        peer,
+                        num_remaining,
+                    })),
+                stats,
+                step,
+            } if num_remaining == 0
+                && self.swarm.behaviour().discovery.health_status == HealthStatus::Initializing =>
+            {
+                warn!(
+                    "Bootstrap query finished but unable to connect to bootnode {:?} {:?}",
+                    stats, step
+                );
+
+                let behaviour = self.swarm.behaviour_mut();
+
+                behaviour.discovery.health_status = HealthStatus::Unhealthy;
+                _ = behaviour
+                    .discovery
+                    .change_interval(self.config.discovery.fast_bootstrap_interval);
             }
 
             Event::OutboundQueryProgressed {
-                result: QueryResult::PutRecord(Err(e)),
                 id,
-                ..
-            } => {
-                error!("PutRecord Failed query_id: {id:?}, error: {e:?}");
+                result:
+                    QueryResult::Bootstrap(BootstrapResult::Ok(BootstrapOk {
+                        peer,
+                        num_remaining,
+                    })),
+                stats,
+                step,
+            } if num_remaining == 0
+                && self
+                    .state_machine
+                    .successfully_connect_to_bootpeer
+                    .is_none()
+                && self.swarm.behaviour().discovery.health_status == HealthStatus::Unhealthy =>
+            {
+                warn!(
+                    "Bootstrap query finished but unable to connect to bootnode {:?} {:?}",
```

Comment:
Same log as above, that could be annoying. Can you change it to reflect the two cases accurately?